packages/ef-runtime-client/src/EFRuntime/EFRuntime.ts

```diff
@@ -0,0 +1,87 @@
+import { IComponentRegistry } from "../ComponentRegistry";
+import { ModuleLoader } from "../ModuleLoader";
+import { StylingHandler } from "../StylingHandler";
+
+export interface IRuntimeDependencies {
+  componentRegistry: IComponentRegistry;
+  moduleLoader: ModuleLoader;
+  stylingHandler: StylingHandler;
+}
+
+export class EFRuntime {
+  private registry: IComponentRegistry;
+  private moduleLoader: ModuleLoader;
+  private stylingHandler: StylingHandler;
+
+  constructor({
+    componentRegistry,
+    moduleLoader,
+    stylingHandler,
+  }: IRuntimeDependencies) {
+    this.registry = componentRegistry;
+    this.moduleLoader = moduleLoader;
+    this.stylingHandler = stylingHandler;
+  }
+
+  private validateOptions(options: {
+    systemCode?: string;
+    overrides?: { [propName: string]: string };
+  }): void {
+    if (!options.systemCode) {
+      throw new Error("Must provide a systemCode option");
+    }
+  }
+
+  async init(
```

Comment:
I don't think we need an `init` function since we have a `constructor` which can be responsible for initializing the Runtime.

Reply:
Hmmm im not sure about this. JavaScript constructors are synchronous, which means you can't await asynchronous operations inside them. Your init methods are asynchronous, so they cannot be moved directly into the constructor. The constructor is responsible for initialising an object's state, while the init methods are responsible for beginning processes that need to be run asynchronously. This separation makes the code easier to test and understand. Keeping init separate allows you to control when these asynchronous operations are triggered, rather than having them run automatically when you create a new object.

Maybe we can change the name? But I still would argue init is appropriate.

Reply:
Yeah, you're right since we are running asyncronous code to fetch the registry and so on. We will need this to be async. 
I agree `init` is appropriate.